src/Ocelot.Provider.Polly/OcelotBuilderExtensions.cs

```diff
@@ -1,115 +1,134 @@
+        where T : class, IPollyQoSResiliencePipelineProvider<HttpResponseMessage>
+        => AddPolly<T>(builder, GetDelegatingHandler, errorMapping);
+
+    /// <summary>
+    /// Add Polly QoS provider to Ocelot
+    /// </summary>
+    /// <typeparam name="T">QOS Provider to use (by default use PollyQoSResiliencePipelineProvider)</typeparam>
+    /// <param name="builder"></param>
+    /// <param name="delegatingHandler">Your customized delegating handler (to manage QOS behavior by yourself)</param>
+    /// <returns></returns>
+    public static IOcelotBuilder AddPolly<T>(this IOcelotBuilder builder, QosDelegatingHandlerDelegate delegatingHandler)
+        where T : class, IPollyQoSResiliencePipelineProvider<HttpResponseMessage>
+        => AddPolly<T>(builder, delegatingHandler, DefaultErrorMapping);
+
+    /// <summary>
+    /// Add Polly QoS provider to Ocelot
+    /// </summary>
+    /// <typeparam name="T">QOS Provider to use (by default use PollyQoSResiliencePipelineProvider)</typeparam>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    public static IOcelotBuilder AddPolly<T>(this IOcelotBuilder builder)
+        where T : class, IPollyQoSResiliencePipelineProvider<HttpResponseMessage>
+        => AddPolly<T>(builder, GetDelegatingHandler, DefaultErrorMapping);
+
+    /// <summary>
+    /// Add Polly QoS provider to Ocelot
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    public static IOcelotBuilder AddPolly(this IOcelotBuilder builder)
+        => AddPolly<PollyQoSResiliencePipelineProvider>(builder, GetDelegatingHandler, DefaultErrorMapping);
+
+    private static DelegatingHandler GetDelegatingHandler(DownstreamRoute route, IHttpContextAccessor contextAccessor, IOcelotLoggerFactory loggerFactory)
+        => new PollyResiliencePipelineDelegatingHandler(route, contextAccessor, loggerFactory);
+
+    #region Obsolete (to remove in a future verison)
+
+    [Obsolete("Use AddPolly instead, it will be remove in future version")]
+    public static IOcelotBuilder AddPollyV7<T>(this IOcelotBuilder builder,
+             QosDelegatingHandlerDelegate delegatingHandler,
+             IDictionary<Type, Func<Exception, Error>> errorMapping)
+             where T : class, IPollyQoSProvider<HttpResponseMessage>
+    {
+        builder.Services
+            .AddSingleton(errorMapping)
+            .AddSingleton<IPollyQoSProvider<HttpResponseMessage>, T>()
+            .AddSingleton(delegatingHandler);
+
+        return builder;
+    }
+
+    [Obsolete("Use AddPolly instead, it will be remove in future version")]
+    public static IOcelotBuilder AddPollyV7<T>(this IOcelotBuilder builder, IDictionary<Type, Func<Exception, Error>> errorMapping)
+        where T : class, IPollyQoSProvider<HttpResponseMessage> =>
+        AddPollyV7<T>(builder, GetDelegatingHandlerV7, errorMapping);
```

Comment:
As far as I understood, `GetDelegatingHandlerV7` is a part of previous version.
Old `AddPolly` method should point to v7 but you've remapped them and introduced breaking changes of force migration to v8!

Reply:
Not an issue! Agreed on force migration to V8